src/community/ogcapi/ogcapi-features/src/main/java/org/geoserver/ogcapi/v1/features/FeatureService.java

```diff
@@ -106,19 +96,38 @@ public class FeatureService {
 
     static final Pattern INTEGER = Pattern.compile("\\d+");
 
-    public static final String CORE = "http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/core";
-    public static final String HTML = "http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/html";
-    public static final String GEOJSON =
-            "http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/geojson";
-    public static final String GMLSF0 =
-            "http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/gmlsf0";
-    public static final String GMLSF2 =
-            "http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/gmlsf2";
-    public static final String OAS30 =
-            "http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/oas30";
+    public static final APIConformance CORE = new APIConformance("http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/core");
+
+    // required
+    public static final APIConformance HTML = CORE.extend("http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/html");
+    public static final APIConformance GEOJSON = CORE.extend("http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/geojson");
+
+    public static final APIConformance GMLSF0 = CORE.extend("http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/gmlsf0");
+    public static final APIConformance GMLSF2 = CORE.extend("http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/gmlsf2)");
+    public static final APIConformance OAS30 = CORE.extend("http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/oas30");
+
+    public static final APIConformance CRS_BY_REFERENCE = CORE.extend("http://www.opengis.net/spec/ogcapi-features-2/1.0/conf/crs");
+
+    public static final APIConformance CQL2_ADVANCED = new APIConformance(ConformanceClass.CQL2_ADVANCED);
+    public static final APIConformance CQL2_ARITHMETIC = new APIConformance(ConformanceClass.CQL2_ADVANCED);
+    public static final APIConformance CQL2_BASIC = new APIConformance(ConformanceClass.CQL2_BASIC);
+    public static final APIConformance CQL2_BASIC_SPATIAL = new APIConformance(ConformanceClass.CQL2_BASIC_SPATIAL);
+    public static final APIConformance CQL2_FUNCTIONS = new APIConformance(ConformanceClass.CQL2_ADVANCED);
+    public static final APIConformance CQL2_PROPERTY_PROPERTY = new APIConformance(ConformanceClass.CQL2_PROPERTY_PROPERTY);
+    public static final APIConformance CQL2_SPATIAL = new APIConformance(ConformanceClass.CQL2_SPATIAL);
+    public static final APIConformance CQL2_TEXT = new APIConformance(ConformanceClass.CQL2_TEXT);
+
+    // informal
+    public static final APIConformance ECQL = new APIConformance(ConformanceClass.ECQL, INFORMAL);
```

Comment:
Vendor, but stable. I believe that's what's important in user eyes, can they rely on the implementation long term, or not? "Informal" does not say, it can be both a vendor experiment (e.g. ogcapi-changesets) or a long term stable feature of the product (e.g., ECQL)

Reply:
Yeah I see your logic; and vendor is consistent with our other extensions.